.storybook/grid/GridPanel.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useEffect, useState, useRef } from 'react';
+import { AddonPanel } from '@storybook/components';
+import { useChannel } from '@storybook/api';
+import { styles } from './gridStyles';
+import { INITIAL_IMAGE_EVENT, IMAGE_SELECTED_EVENT } from './withGrid';
+import { STORY_CHANGED } from '@storybook/core-events';
+import { IframePostMessageService } from '@guardian/grid-client';
+import { Reporter } from '@guardian/grid-client/lib/utils';
+
+const GRID_URL =
+    process.env.NODE_ENV === 'development'
+        ? 'https://media.test.dev-gutools.co.uk'
+        : 'https://media.gutools.co.uk';
+
+const isValidMessage = (data) => data?.crop?.data && data?.image?.data;
+
+export const GridPanel = ({ api, active }) => {
+    const emit = useChannel({});
+    const [isModalOpen, setIsModalOpen] = useState(false);
+    const [selectedImage, setSelectedImage] = useState();
+    const activeButton = useRef(null);
+
+    const onMessage = (event) => {
+        if (event.origin !== GRID_URL) {
+            return;
+        }
+
+        const postMessageService = new IframePostMessageService(event, Reporter.default);
+
+        if (!postMessageService.isValid) {
+            return;
+        }
+
+        const cropAssets = event.data.crop.data.assets;
+        const cropUrl = cropAssets[cropAssets.length - 1]?.secureUrl;
```

Comment:
Won't this code break if the image has no crops (which is somewhat common)?

Reply:
The `isValidMessage` function checks for a crop, but doesn't seem to be used. Once you have a crop, it *should* have assets available, but maybe we want to double check with EdTools that this is true - a crop will always have at least one asset. If not, we can add this check to `isValidMessage` (once that validation is used).

Reply:
We've replaced our own homegrown `isValidMessage` with `!postMessageService.isValid` from `@guardian/grid-client` which, if I understand correctly, will return `false` if there are no crops. I'll double check that though, it'd be good to be sure. I'd say if the validation _isn't_ right it'd be good to capture those rules in `guardian/grid-client` rather than expanding on them here.